src/main/java/com/google/firebase/internal/FirebaseThreadManagers.java

```diff
@@ -84,7 +84,8 @@ private static class DefaultThreadManager extends GlobalThreadManager {
     protected ExecutorService doInit() {
       ThreadFactory threadFactory = FirebaseScheduledExecutor.getThreadFactoryWithName(
           getThreadFactory(), "firebase-default-%d");
-      return Executors.newCachedThreadPool(threadFactory);
+
+      return Executors.newFixedThreadPool(100, threadFactory);
     }
```

Comment:
Yes the downside of the this is that threads once created aren't closed until shutdown. 

Made a change to add a keepAliveTime for idle threads so this would no longer be the case. I was missing the `allowCoreThreadTimeOut()` method in testing which allows the timer to be applied.

Reply:
Thanks! This LGTM!